pkg/util/test/kms_test.go

```diff
@@ -0,0 +1,136 @@
+package kms_test
+
+import (
+	"os"
+	"time"
+
+	nbv1 "github.com/noobaa/noobaa-operator/v5/pkg/apis/noobaa/v1alpha1"
+	"github.com/noobaa/noobaa-operator/v5/pkg/options"
+	"github.com/noobaa/noobaa-operator/v5/pkg/system"
+	"github.com/noobaa/noobaa-operator/v5/pkg/util"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func getMiniNooBaa() *nbv1.NooBaa {
+	options.MiniEnv = true
+	options.Namespace = corev1.NamespaceDefault
+	nb := system.LoadSystemDefaults()
+	return nb
+}
+
+func nooBaaCondStatus(noobaa* nbv1.NooBaa, s corev1.ConditionStatus) bool {
+	found := false
+
+	timeout := 120 // seconds
+	for i := 0; i < timeout; i++ {
+		_, _, err := util.KubeGet(noobaa)
+		Expect(err).To(BeNil())
+
+		if kmsStatus(noobaa, s) {
+			found = true
+			break
+		}
+		time.Sleep(time.Second)
+	}
+
+	return found
+}
+
+func simpleKmsSpec(token, api_address string) nbv1.KeyManagementServiceSpec {
+	kms := nbv1.KeyManagementServiceSpec{}
+	kms.TokenSecretName = token
+	kms.ConnectionDetails = map[string]string{
+		util.VaultAddr : api_address,
+		util.VaultBackendPath : "noobaa/",
```

Comment:
add another scenario with KV V2, it's important since we know there were issues with libopenstorage/secrets with kv v2

Reply:
See #797, [commit](https://github.com/noobaa/noobaa-operator/pull/797/commits/e9713024359d59eded6c4fc85b8fdef7dd23a8e6)